obspy/clients/filesystem/sds.py

```diff
@@ -0,0 +1,165 @@
+    "{network}.{station}.{location}.{channel}.{type}.{year}.{doy:03d}")
+
+
+class Client(object):
+    """
+    Request client for SeisComP Data Structure archive on local filesystem.
+
+    For details see the :meth:`~obspy.clients.filesystem.sds.Client.__init__()`
+    method.
+    """
+    def __init__(self, sds_root, sds_type="D", format="MSEED",
+                 fileborder_seconds=30, fileborder_samples=5000):
+        """
+        Initialize a SDS local filesystem client.
+
+        >>> from obspy.clients.filesystem.sds import Client
+        >>> client = Client("/my/SDS/archive/root")  # doctest: +SKIP
+
+        :type sds_root: str
+        :param sds_root: Root directory of SDS archive.
+        :type sds_type: str
+        :param sds_type: SDS data type identifier, one single character. Types
+            recommended by the SDS definition are: 'D' - Waveform data,
+            'E' - Detection data, 'L' - Log data, 'T' - Timing data,
+            'C' - Calibration data, 'R' - Response data, 'O' - Opaque data. Can
+            also be wildcarded by setting to `?` or `*`.
+        :type format: str
+        :param format: File format the data is stored in, see
+            :func:`~obspy.core.stream.read()` for a list of file formats
+            supported by ObsPy. Usually, SDS archives are stored in "MSEED"
+            format. Can be set to `None` for file format autodetection (slowing
+            down the reading).
+        """
+        if not os.path.isdir(sds_root):
+            msg = ("SDS root is not a local directory: " + sds_root)
+            raise IOError(msg)
+        self.sds_root = sds_root
+        self.sds_type = sds_type
+        self.format = format
+        self.fileborder_seconds = fileborder_seconds
+        self.fileborder_samples = fileborder_samples
+
+    def get_waveforms(self, network, station, location, channel, starttime,
+                      endtime, cleanup_merge=True, sds_type=None):
+        """
+        Read data from a local SeisComP Data Structure (SDS) directory tree.
+
+        >>> from obspy import UTCDateTime
+        >>> t = UTCDateTime("2015-10-12T12")
+        >>> st = client.get_waveforms("IU", "ANMO", "*", "HH?", t, t+30)
+        ... # doctest: +SKIP
+
+        :type seed_id: str
+        :param seed_id: SEED ID of data that should be read
+            (e.g. 'IU.ANMO.00.LHZ'). Wildcards '*' and '?' are supported
+            (e.g. 'IU.*.*.HH?').
+        :type starttime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param starttime: Start of requested time window.
+        :type endtime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param endtime: End of requested time window.
+        :type cleanup_merge: bool
+        :param cleanup_merge: Whether to perform a cleanup merge on the
+            resulting stream to merge seamless traces originating from
+            different files (:meth:`st.merge(-1)
+            <obspy.core.stream.Stream.merge>`)
```

Comment:
It might be useful to be able to set the merge strategy here, e.g. `False` is not merge, any numerical value would correspond to one of the merge methods.